framework/src/meshgenerators/ModifyNodeGenerator.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ModifyNodeGenerator.h"
+
+registerMooseObject("MooseApp", ModifyNodeGenerator);
+
+InputParameters
+ModifyNodeGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>("input", "Input mesh to modify");
+
+  params.addRequiredParam<std::vector<int>>("node_id", "Id of modified node");
+
+  params.addRequiredParam<std::vector<Point>>("new_position", "New position in vector space");
+
+  params.addClassDescription("Modifies the position of one node");
+
+  return params;
+}
+
+ModifyNodeGenerator::ModifyNodeGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _node_id(getParam<std::vector<int>>("node_id")),
+    _new_position(getParam<std::vector<Point>>("new_position"))
+{
+}
+
+std::unique_ptr<MeshBase>
+ModifyNodeGenerator::generate()
+{
+  if (_node_id.size() != _new_position.size())
+    mooseError("Node ids and new positions sizes do not match. Please check your input");
```

Comment:
Move this up into the object constructor (checking earlier is better). In addition, use a `paramError` so that it provides some context:

```
if (_node_id.size() != _new_position.size())
  paramError("node_id", "Does not match the length of 'new_position'");
```